flexget/plugins/plugin_sftp.py

```diff
@@ -0,0 +1,423 @@
+        except KeyError:
+            private_key_pass = None
+
+        config = ConenctionConfig(host, port, username, password, private_key, private_key_pass)
+
+        return config
+
+    def prepare_config(self, config, task):
+        """
+        Sets defaults for the provided configuration
+        """
+        config.setdefault('to', os.path.join(task.manager.config_base, 'temp'))
+
+        return config
+
+    def download_entry(self, entry, config, sftp):
+        """
+        Downloads the file(s) described in entry
+        """
+
+        path = urlparse(entry['url']).path or '.'
+        delete_origin = config['delete_origin']
+
+        to = config['to']
+        if to:
+            try:
+                to = render_from_entry(to, entry)
+            except RenderError as e:
+                log.error('Could not render path: %s' % to)
+                entry.fail(e)
+                return
+
+        if not sftp.lexists(path):
+            log.error('Remote path does not exist: %s' % path)
+            return
+
+        if not os.path.exists(to):
+            os.makedirs(to)
+
+        os.chdir(to)
+
+        if sftp.isdir(path):
+            base_path = os.path.normpath(os.path.join(path, '..'))
+            dir_name = os.path.basename(path)
+
+            log.verbose('Downloading directory %s' % path)
+            try:
+                sftp.cwd(base_path)
+                sftp.get_r(dir_name, to)
+            except Exception as e:
+                error = 'Failed to download directory %s (%s)' % (path, e)
+                log.error(error)
+                entry.fail(error)
+                return
+
+            if delete_origin:
+                log.debug('Attempting to delete directory %s' % path)
+                try:
+                    sftp.rmdir(dir_name)
+                except Exception as e:
+                    log.error('Failed to delete directory %s' % e)
+                    return
+
+        elif sftp.isfile(path):
+            destination = os.path.join(to, os.path.basename(path))
+
+            log.verbose('Downloading file %s' % path)
+            try:
+                sftp.get(path, destination)
+            except Exception as e:
+                error = 'Failed to download file %s (%s)' % (path, e)
+                log.error(error)
+                entry.fail(error)
+                return
+
+            if delete_origin:
+                try:
+                    sftp.remove(path)
+                except Exception as e:
+                    log.error('Failed to delete file %s (%s)' % (path, e))
+                    return
+        else:
+            log.warn('Skipping unknown file %s' % path)
+
+    def on_task_download(self, task, config):
+        """
+        Task handler for sftp_download plugin
+        """
+        dependency_check()
+        config = self.prepare_config(config, task)
+
+        # Download entries by host so we can reuse the connection
```

Comment:
Should we be filtering to only try to process entries using the sftp uri scheme?